### ISSUE TITLE
webpack-cli@4 does not exit with non-0 code on error

### ISSUE THREAD

**Issue by fregante** [webpack/webpack#11789]
# Bug report

`$ webpack` should exit with a non-0 code when it finds an error in `webpack.config.js`

**What is the current behavior?**

The `webpack` version was also changed between screenshots, but it doesn't actually affect the result. The same thing happens with `webpack@4` and `webpack-cli@4`

<img width="237" alt="" src="https://user-images.githubusercontent.com/1402241/96945484-834cb800-14a3-11eb-8eb1-0605f5eb9d4d.png">

**If the current behavior is a bug, please provide the steps to reproduce.**

```sh
npm install -g webpack webpack-cli@4
echo '.' > webpack.config.js
webpack
```

**What is the expected behavior?**

`webpack-cli@3` worked correctly, it exited with `1`

<img width="241" alt="" src="https://user-images.githubusercontent.com/1402241/96945472-7cbe4080-14a3-11eb-80e3-46e9733c0781.png">

**Other relevant information:**
webpack version: 4 or 5 (webpack version irrelevant)
webpack-cli version: 4.1.0
Node.js version: v13.8.0
Operating System: macOS 10.15.4
Additional tools:

**Comment by snitin315** [comment]
Thanks for the report I'm looking into it.

**Comment by snitin315** [comment]
Working fine for me. 

![Screenshot at 2020-10-23 07-47-54](https://user-images.githubusercontent.com/46647141/96948833-21706a80-1504-11eb-8dec-ffd23a9f6d06.png)

**Comment by fregante** [comment]
I don't know where you're testing because it doesn't work anywhere for me. Here's a plain GitHub Action workflow where CI passes even if webpack fails: https://github.com/fregante/webpack-issue-11789/runs/1296100000?check_suite_focus=true

```yml
name: CI
on: [push]
jobs:
  build:
    runs-on: ubuntu-latest
    steps: 
      - run: npm install webpack webpack-cli@4
      - run: echo '.' > webpack.config.js
      - run: npx webpack
```

<img width="564" alt="" src="https://user-images.githubusercontent.com/1402241/96953506-d3814580-14b6-11eb-8b1b-3bf70481a941.png">

**Comment by snitin315** [comment]
Just added a test case for the same in webpack/webpack-cli#1979

**Comment by fregante** [comment]
Yes but the test passes while the example code doesn’t work on GitHub Actions and on my computer, so the test isn’t effectively testing the same thing/environment.

**Comment by fregante** [comment]
This is the same demo code running on webpack-cli@3: https://github.com/fregante/webpack-issue-11789/runs/1297071918?check_suite_focus=true

It fails as expected.

**Comment by fregante** [comment]
Here’s another workflow testing every major version of webpack-cli. Every version correctly fails except v4

https://github.com/fregante/webpack-issue-11789/runs/1297087059?check_suite_focus=true

**Comment by fregante** [comment]
It's possible that https://github.com/webpack/webpack-cli/pull/1971 fixed it just a couple of days ago but hasn't been pushed yet 🍰

**Comment by snitin315** [comment]
can you try with the `master` branch?

**Comment by fregante** [comment]
I tried but it's a monorepo, I don't know how to `npm install` that

**Comment by snitin315** [comment]
You can create a symlink with `npm link webpack-cli`
